commit 9835db39bb8151cf4471a792a878318c8d07bf4d
Author: Hannes Reinecke <hare@suse.de>
Date:   Fri Mar 18 14:55:38 2016 +0100

    scsi_common: do not clobber fixed sense information
    
    commit ba08311647892cc7912de74525fd78416caf544a upstream.
    
    For fixed sense the information field is 32 bits, to we need to truncate
    the information field to avoid clobbering the sense code.
    
    Fixes: a1524f226a02 ("libata-eh: Set 'information' field for autosense")
    Signed-off-by: Hannes Reinecke <hare@suse.com>
    Reviewed-by: Lee Duncan <lduncan@suse.com>
    Reviewed-by: Bart Van Assche <bart.vanassche@sandisk.com>
    Reviewed-by: Ewan D. Milne <emilne@redhat.com>
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/scsi/scsi_common.c b/drivers/scsi/scsi_common.c
index c126966..ce79de8 100644
--- a/drivers/scsi/scsi_common.c
+++ b/drivers/scsi/scsi_common.c
@@ -278,8 +278,16 @@ int scsi_set_sense_information(u8 *buf, int buf_len, u64 info)
 		ucp[3] = 0;
 		put_unaligned_be64(info, &ucp[4]);
 	} else if ((buf[0] & 0x7f) == 0x70) {
-		buf[0] |= 0x80;
-		put_unaligned_be64(info, &buf[3]);
+		/*
+		 * Only set the 'VALID' bit if we can represent the value
+		 * correctly; otherwise just fill out the lower bytes and
+		 * clear the 'VALID' flag.
+		 */
+		if (info <= 0xffffffffUL)
+			buf[0] |= 0x80;
+		else
+			buf[0] &= 0x7f;
+		put_unaligned_be32((u32)info, &buf[3]);
 	}
 
 	return 0;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1442_9835db39bb8151cf4471a792a878318c8d07bf4d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1442_9835db39bb8151cf4471a792a878318c8d07bf4d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
